packages/@uppy/box/src/Box.tsx

```diff
@@ -71,9 +71,15 @@ export default class Box<M extends Meta, B extends Body>
     )
     this.provider = new Provider(uppy, {
       companionUrl: this.opts.companionUrl,
-      companionHeaders: this.opts.companionHeaders,
-      companionKeysParams: this.opts.companionKeysParams,
-      companionCookiesRule: this.opts.companionCookiesRule,
+      ...(this.opts.companionHeaders !== undefined && {
```

Comment:
for example these we could also simplify
```suggestion
      ...(this.opts.companionHeaders && {
```